### PR TITLE
perf: add encrypted key api

### PR DESCRIPTION
**Pull request by fit2bot**
perf: add encrypted key api

### CODE REVIEW

File: pkg/koko/koko.go [hunk 1]
```diff
@@ -55,9 +56,31 @@ func RunForever(confPath string) {
...
+			cfg.UpdateRedisPassword(ret.Value)
+		} else {
+			logger.Error("Get encrypted config value failed: empty value")
+		}
+	}
+}
+
 func runTasks(jmsService *service.JMService) {
 	if config.GetConf().UploadFailedReplay {
 		go uploadRemainReplay(jmsService)
```
Comment (BaiJiangJie) [inline pkg/koko/koko.go:86]: Unfortunately, you haven't provided the actual code that needs to be checked for differences, improvements or optimizations. Please share the relevant sections of the code so I can assist with analyzing them.

Here are general guidelines on how one might approach checking code:

1. `Review:` Look at the structure and format of your code to ensure it is well-documented, easy for other developers to read.
   
2. `Analyse:`
   - Compare function arguments, parameters, return values, etc.
   - Check if there have been any changes since last analysis.
   - Review if any functions have been moved/renamed/replaced
   - Analyze error handling practices

Optimization Suggestion:
For any potential performance improvement in terms of code clarity, readability, efficiency, etc., please provide some sample context from which this advice could be derived (for example, specific lines needing optimization).

Suggestions for code difference detection:
Consider reviewing the following elements:

- If there's a change in syntax used (like using new keyword instead of define)

- Consider comparing function bodies too. For instance, compare if an existing function name has changed while retaining all functionality

In summary, without specifics about what needs verification, it'll be hard to give accurate answers.

File: pkg/jms-sdk-go/service/jms_terminal.go [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+		"secret_encrypt_key": encryptKey,
+		"encrypted_value":    encryptedValue,
+	}
+	_, err = s.authClient.Post(TerminalEncryptedConfigURL, data, &resp)
+	return
+}
+
+type ResultValue struct {
+	Value string `json:"value"`
+}
```
Comment (BaiJiangJie) [inline pkg/jms-sdk-go/service/jms_terminal.go:14]: The code you provided has been updated to remove the `Package` and `service`, which should be removed as these lines are not used anymore.

Here's what's changed:

1) The function name "GetEncryptedConfigValue" is now correctly written with lower case letters (`getEncryptedConfigValue`) instead of uppercase ("GET_ENCRYPTEDCONFIGVALUE").
 
No changes were made to the code structure or logic due to no apparent errors or issues being detected. No further modifications are needed at this point unless more specific issues need addressing.

Therefore, there are no major differences or irregularities in current version of your service code snippet. If there are still minor adjustments that could improve performance or adhere better to best practices, feel free to ask for details on those areas!

File: pkg/config/config.go [hunk 3]
```diff
@@ -82,6 +84,10 @@ func (c *Config) EnsureConfigValid() {
 	}
 }
 
+func (c *Config) UpdateRedisPassword(val string) {
+	c.RedisPassword = val
+}
+
 func GetConf() Config {
 	if GlobalConfig == nil {
 		return getDefaultConfig()
```
Comment (BaiJiangJie) [inline pkg/config/config.go:93]: There is a slight difference between the two versions of the code you provided. The first version contains an empty string literal `SecretEncryptKey` which could lead to some unexpected behavior if used elsewhere. In this case, it's not specified what it means exactly in the context of `config`, so I would recommend using a more precise name instead (e.g., SecretEncryptionKey). 

Additionally, there was no previous implementation for RedisPassword being set in UpdateRedisPassword function.

This small refinement can help avoid any possible bugs or confusion related to these changes.

**Review by LeeEirc — APPROVED** [review]
